packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts

```diff
@@ -455,7 +455,7 @@ export class OdspDocumentDeltaConnection extends DocumentDeltaConnection {
     protected disconnectHandler = (error: IFluidErrorBase & OdspError, clientId?: string) => {
         if (clientId === undefined || clientId === this.clientId) {
             this.disconnect(error);
-            this.logger.sendTelemetryEvent({ eventName: "ServerDisconnect", clientId: this.clientId }, error);
+            this.logger.sendTelemetryEvent({ eventName: "ServerDisconnect", clientId: this.hasDetails ? this.clientId : undefined }, error);
```

Comment:
Can you please change the order of line 457 and this i.e. send telemetry event before calling this.disconnect(error).